### PR TITLE
Provide interface to validate the entire AST

### PR DESCRIPTION
**Pull request by qinezh**

### CODE REVIEW

**Comment by vicancy** [comment]
>         public void LoadEnabledRulesProvider()

why is it `public`? expecting anyone from outside call this?

---
Refers to: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Validation/MarkdownValidatorBuilder.cs:221 in b3ad091. [](commit_id = b3ad091c707abd2134e3d4a2929bccf78000ab5b, deletion_comment = False)

File: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Validation/MarkdownValidatorBuilder.cs [hunk 1]
```diff
@@ -71,6 +73,13 @@ public IMarkdownObjectRewriter CreateRewriter()
                 MarkdownValidatePhaseName);
         }
 
+        public ImmutableArray<IMarkdownDocumentValidator> CreateMarkdownDocumentValidators()
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Validation/MarkdownValidatorBuilder.cs]: >CreateMarkdownDocumentValidators [](start = 58, length = 32)

Is it needed? Is it enough to expose `Create` only?

**Comment by 928PJY** [comment]
>         public void LoadEnabledRulesProvider()

It is called in test, please set it as `internal` and set `InternalsVisibleTo` in assemblyInfo.cs

---
In reply to: [374103250](https://github.com/dotnet/docfx/pull/2529#issuecomment-374103250) [](ancestors = 374103250)

---
Refers to: src/Microsoft.DocAsCode.MarkdigEngine.Extensions/Validation/MarkdownValidatorBuilder.cs:221 in b3ad091. [](commit_id = b3ad091c707abd2134e3d4a2929bccf78000ab5b, deletion_comment = False)

**Comment by qinezh** [comment]
Thanks @vicancy for the suggestion. I closed this one and submit a new PR here: https://github.com/dotnet/docfx/pull/2530

**Pull request closed without merging**